packages/@lwc/template-compiler/src/parser/attribute.ts

```diff
@@ -180,16 +180,17 @@ export function isValidTabIndexAttributeValue(value: any): boolean {
     return value === '0' || value === '-1';
 }
 
-export function isAriaOrDataOrFmkAttribute(attrName: string): boolean {
-    return isAriaAttribute(attrName) || isFmkAttribute(attrName) || isDataAttribute(attrName);
+export function isAriaOrDataOrFrameworkAttribute(attrName: string): boolean {
+    return isAriaAttribute(attrName) || isFrameworkAttribute(attrName) || isDataAttribute(attrName);
 }
 
 function isDataAttribute(attrName: string): boolean {
     return !!attrName.match(DATA_RE);
 }
 
-function isFmkAttribute(attrName: string): boolean {
-    return attrName === 'key' || attrName === 'slot';
+function isFrameworkAttribute(attrName: string): boolean {
+    // 'key' is currently the only LWC framework-specific attribute that doesn't start with "lwc:"
+    return attrName === 'key';
```

Comment:
There is no need to put `slot` here because it's already in the list of global HTML attributes, and `isFrameworkAttribute` is only used in one place, which is the `isValidHTMLAttribute` check, which already checks the list of global HTML attributes.